src/server/services/checkWebsiteStatus.js

```diff
@@ -0,0 +1,52 @@
+const cron = require('node-cron');
+const { getConnection } = require('../db');
+const { logMailer } = require('../logMailer');
+const LogEmail = require('../models/LogEmail');
+const colors = require('colors');
+  
+async function checkWebsite() {
+  const conn = getConnection();
```

Comment:
I agree with you to move conn after verify website status.

But I believe I still need a conn to pass into logMail method as parameter

Reply:
See other comment about not directly calling logMail. If you use the OED standard logging method then no conn is needed as it takes care of this. It might help with this but let me know if not so we can figure out what to do.

Reply:
is the OED standard logging method located in 'sendLogEmail.js' ?? could you direct me to the file?

Reply:
Sorry. Check out src/server/log.js. It will print to console, add to log email, add date and add to log.txt (soon to go into DB). The code normally calls the info, warn, error methods that call the main one.

Reply:
I see. Thank you for bringing this to my attention. 
I can picture how the script will look like using the existing OED functions. I will list out all the steps below and correct me if im wrong or if there is anything to change. I hope to get this merged by Monday 12/9 so I can showcase our work for my college :)

Step 1:
create a new file containing website checking function named 'checkWebsiteStatus.js' 

Step 2:
use js fetch function to verify website status by sending a head request. 

Step 3:
if website is down, call the function in log.js to start database connection and log it in the database

Step 4:
call function in sendLogEmail.js to send email to all users 

Step 5:
insert an object in package.json under 'scripts', with key being 'checkWebsiteStatus' and value being the relative path to 'checkWebsiteStatus.js' 

Step 6:
create a new bash file named 'checkWebsiteStatusCron.bash' containing the similiar format as below and replace 'sendLogEmail' with 'checkWebsiteStatus'

<img width="888" alt="image" src="https://github.com/user-attachments/assets/5c4ef6ac-a2a4-42dd-ba18-82f37051b26d">

Step 7:
user can run the following cron command to set up hourly check

 chmod +x /etc/cron.hourly/checkWebsiteStatusCron.bash

Reply:
First, good luck with your showcase. You are welcome to show whatever you have done even if it is not yet merged. Having said that, we can continue to push to get this completed soon.

My only comment right now is steps 3 & 4. If the code calls the error function in src/server/log.js then it will log to appropriate places (including the upcoming change to support an admin page to look at log message) and prepare the message to send via email if appropriate. I don't think you need to do step 4 because properly doing step 3 will cover that. Does this make sense?

Reply:
You are correct. I got myself confused since I didnt see 'sendLogEmail' function being implemented, but it is running as a cron job to check the database for appropriate message and send.

That makes sense. I will skip step 4 then.

Reply:
Just submitted another pull request with the changes we discussed. The commands for user to execute the cron job and run the scripts are as below:

chmod +x /etc/cron.hourly/checkWebsiteStatusCron.bash

/etc/cron.hourly/checkWebsiteStatusCron.bash www.example.com